subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -20,13 +20,30 @@
 
 LOG_MODULE_DECLARE(bt_ots, CONFIG_BT_OTS_LOG_LEVEL);
 
+/* In order to transmit the maximum size we choose CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU
+ * as the buffer size. However, directory listing encoding can only happen at
+ * directory entry boundary so we must ensure the MTU value is rounded up to the
+ * nearest directory entry size to not overflow our buffer. Also since we can only
+ * encode at object entry boundary, we need to make sure we have at least room for
+ * two entries so entries can be read from the middle.
+ */
```

Comment:
I don't understand this comment very well. 

> directory listing encoding can only happen at directory entry boundary so we must ensure the MTU value is rounded up to the nearest directory entry size to not overflow our buffer

What does it mean that it can only happen at directory entry boundary?

> Also since we can only encode at object entry boundary, we need to make sure we have at least room for two entries so entries can be read from the middle.

I don't follow why we will need 2 entries. Can you elaborate?

Reply:
I will try to reword the comments and add some ASCII art to make it more obvious.

Reply:
As I tried to explain this, I realized I can simplify this more, so hopefully the new version makes more sense.